### PR TITLE
Properly expose errors in ads and tests

### PR DESCRIPTION
**Pull request by anthonydresser**
- Zonejs was taking over errors in ads and avoiding vscode error handling. This fixes that by bypassing zonejs for unhandled promises.
- For different reasons we also had errors in our tests that were not failing the tests, so I added an unhandled promise handler to the test framework and rewrote the structure of some of the tests. I found few (~10) failing tests and fixes them, and 1 actual product behavior issue that I fixed.

### CODE REVIEW

File: src/sql/setup.js [hunk 1]
```diff
@@ -22,4 +22,7 @@ define(["require", "exports"], function (require) {
 	require.__$__nodeRequire('zone.js/dist/zone-error');
 	require.__$__nodeRequire('chart.js');
 	window["Zone"]["__zone_symbol__ignoreConsoleErrorUncaughtError"] = true;
+	window["Zone"]["__zone_symbol__unhandledPromiseRejectionHandler"] = e => setImmediate(() => {
```
Comment (Charles-Gagnon) [inline src/sql/setup.js]: Out of curiosity - where are finding these values such as `__zone_symbol__unhandledPromiseRejectionHandler`? I did a quick search and the only things coming up are PRs like this one 😄 Are you just looking through the source or something?

File: src/sql/platform/accounts/common/accountViewModel.ts [hunk 2]
```diff
@@ -57,7 +58,7 @@ export class AccountViewModel {
 								},
 								() => { /* Swallow failures at getting accounts, we'll just hide that provider */ });
 					});
-					return Promise.all(promises);
+					return Promise.all(promises).then(accounts => coalesce(accounts));
```
Comment (Charles-Gagnon) [inline src/sql/platform/accounts/common/accountViewModel.ts]: I assume this was the behavior fix?

Comment (anthonydresser) [reply src/sql/platform/accounts/common/accountViewModel.ts]: Yes, before if the getAccountsForProvider rejected, we were swallowing the error and not returning anything which would become an undefined value in the Promise.all response array.

File: src/sql/setup.js (hunk 1 above)
Comment (anthonydresser) [reply src/sql/setup.js]: Looking through the source code of the library

**Pull request merged**